gadgets/src/bits/boolean_input.rs

```diff
@@ -255,6 +236,40 @@ impl<F: PrimeField, CF: PrimeField> FromFieldElementsGadget<F, CF> for BooleanIn
     }
 }
 
+impl<F: PrimeField, CF: PrimeField> MergeGadget<CF> for BooleanInputGadget<F, CF> {
+    fn merge<CS: ConstraintSystem<CF>>(&self, _cs: CS, other: &Self) -> Result<Self, SynthesisError> {
+        let mut elems = vec![];
```

Comment:
Mark.

Reply:
note: the vectors that are grown using `extend_from_slice` usually don't need pre-allocating, as they are elongated "in bulk", so the number of reallocations is limited to the number of `extend_from_slice` calls.